azure/services/managedclusters/spec.go

```diff
@@ -365,6 +363,20 @@ func (s *ManagedClusterSpec) Parameters(existing interface{}) (params interface{
 			}
 			existingMC.AddonProfiles[key].Identity = nil
 		}
+
+		if s.UserAssignedIdentities != nil {
```

Comment:
The if is redundant code at line number 330 is also doing the same.

Reply:
Removed this code.